### PR TITLE
Introducing E2E tests in GNS3 Web UI

### PR DESCRIPTION
**Pull request by piotrpekala7**
**With this pr E2E tests for all major functionalities of GNS3 Web UI should be delivered**

### CODE REVIEW

**Comment by piotrpekala7** [comment]
@ziajka Can we list all functionalities that should be covered?
Can you add sth to the list below?
1. ability to find and add server
2. ability to create new project
3. ability to add new nodes to project
4. ability to manage GNS3 settings
5. ability to manage template preferences

**Comment by ziajka** [comment]
Looks good. We may want to have E2E tests in the build pipeline.

**Comment by piotrpekala7** [comment]
I went through https://angular.io/guide/styleguide (...folder and file structure...). There is no recommended place to locate e2e's

File: e2e/servers.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        // assert
+        expect(text).toBe("We've discovered GNS3 server on 127.0.0.1:3080, would you like to add to the list?");
+    });
+
+    it('user should see added server in the list', async () => {
+        // arrange
+        page.maximizeWindow();
+        await page.navigateToServersPage();
+        await page.clickAddServer();
+        helper.sleep(1000);
```
Comment (piotrpekala7) [inline e2e/servers.e2e-spec.ts]: @ziajka Is there any way to avoid browser sleep here?

**Pull request merged**